src/ring_theory/polynomial/chebyshev/defs.lean

```diff
@@ -113,3 +120,91 @@ end
+|0        := by simp only [chebyshev₂_zero, chebyshev₂_one, chebyshev₁_one, two_mul, mul_one]
+|1        := by simpa only [chebyshev₂_one, chebyshev₁_two, chebyshev₂_two, ←mul_assoc, pow_two, mul_comm, add_sub, ←mul_add]
+|(n + 2)  := begin
+  calc chebyshev₂ R (n + 2 + 1) = 2 * X * chebyshev₂ R (n + 1 + 1) - chebyshev₂ R (n + 1) : by rw chebyshev₂_add_two
+  ... = 2 * X * (X * chebyshev₂ R (n + 1) + chebyshev₁ R (n + 2)) - (X * chebyshev₂ R n + chebyshev₁ R (n+1)) : by simp only [chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁ n, chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁ (n+1)]
+  ... = (2 * X * (X * chebyshev₂ R (n + 1)) + 2 * X * chebyshev₁ R (n+2)) - X * chebyshev₂ R n - chebyshev₁ R (n+1) : by rw [mul_add, sub_add_eq_sub_sub]
+  ... = (2 * X * (X * chebyshev₂ R (n + 1))) - X * chebyshev₂ R n + 2 * X * chebyshev₁ R (n+2) - chebyshev₁ R (n+1) : by simp only [sub_add_eq_add_sub]
+  ... =  X * (2 * (X * chebyshev₂ R (n + 1)) - chebyshev₂ R n) + (2 * X * chebyshev₁ R (n + 1 + 1) - chebyshev₁ R (n + 1)) : by simp only [add_sub_assoc, mul_comm, mul_assoc, mul_sub]
+  ... = X * chebyshev₂ R (n + 2) + chebyshev₁ R (n + 2 + 1) : by simp only [mul_assoc, chebyshev₂_add_two, chebyshev₁_add_two]
+end
+
+lemma chebyshev₁_eq_chebyshev₂_sub_X_mul_chebyshev₂ :
+∀ (n : ℕ), chebyshev₁ R (n+1) = chebyshev₂ R (n+1) - X * chebyshev₂ R n :=
+begin
+  intro n,
+  calc chebyshev₁ R (n+1) = chebyshev₁ R (n+1) + X * chebyshev₂ R n - X * chebyshev₂ R n : by rw add_sub_cancel
+  ...                     = X * chebyshev₂ R n + chebyshev₁ R (n+1) - X * chebyshev₂ R n : by rw add_comm
+  ...                     = chebyshev₂ R (n+1) - X * chebyshev₂ R n: by rw chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁
+end
+
+lemma chebyshev₁_eq_X_mul_chebyshev₁_sub_pol_chebyshev₂ :
+∀ (n : ℕ), chebyshev₁ R (n+2) = X * chebyshev₁ R (n+1) - (1 - X ^ 2) * chebyshev₂ R n
+|0        := by simp only [chebyshev₁_one, chebyshev₁_two, chebyshev₂_zero, mul_one,
+                 sub_sub_assoc_swap, pow_two, two_mul]
+|1        := begin simp only [chebyshev₁_add_two, chebyshev₁_zero, chebyshev₁_add_two, chebyshev₂_one, chebyshev₁_one, sub_mul, mul_sub, mul_one, one_mul], sorry end
+|(n + 2)  := begin rw chebyshev₁_add_two, sorry end
+
+variables {R S}
+
+lemma map_chebyshev₂ (f : R →+* S) :
+  ∀ (n : ℕ), map f (chebyshev₂ R n) = chebyshev₂ S n
+| 0       := by simp only [chebyshev₂_zero, map_one]
+| 1       := begin simp only [chebyshev₂_one, map_X, map_mul, map_add, map_one],
+                   change map f (1+1) * X = 2 * X,
+                   simp only [map_add, map_one],
+                   refl end
+| (n + 2) :=
+begin
+  simp only [chebyshev₂_add_two, map_mul, map_sub, map_X, bit0, map_add, map_one],
+  rw [map_chebyshev₂ (n + 1), map_chebyshev₂ n],
+end
+
+lemma chebyshev₂_derivative_eq_chebyshev₁ :
+∀ (n : ℕ), derivative (chebyshev₁ R (n + 1)) = (n + 1) * chebyshev₂ R n
+|0        := begin simp only [chebyshev₁_one, chebyshev₂_zero, derivative_X], norm_num end
```

Comment:
Note that if you invoke only one tactic, then you can use `by tac` instead of `begin tac end`.

Reply:
I know (at least in theory). Here there is an invocation of norm_num at the end though.

Reply:
Oops... I should learn to read properly :facepalm: 
Anyway, there is also `by { tac1, tac2 }` for what it's worth.